src/psyclone/tests/psyGen_test.py

```diff
@@ -2621,8 +2631,465 @@ def test_binaryoperation_can_be_printed():
+    assert "Assignment" in str(kschedule)  # Check children are printed
+    assert "End Schedule" in str(kschedule)
+
+
+# Test Symbol Class
+def test_symbol_initialization():
+    '''Test that a Symbol instance can be created when valid arguments are
+    given, otherwise raise relevant exceptions.'''
+
+    # Test with valid arguments
+    assert isinstance(Symbol('a', 'real', [], 'local'), Symbol)
+    assert isinstance(Symbol('a', 'integer', [], 'local'), Symbol)
+    assert isinstance(Symbol('a', 'real', [], 'read_arg'), Symbol)
+    assert isinstance(Symbol('a', 'real', [], 'external'), Symbol)
+    assert isinstance(Symbol('a', 'real', [], 'write_arg'), Symbol)
+    assert isinstance(Symbol('a', 'real', [], 'readwrite_arg'), Symbol)
+    assert isinstance(Symbol('a', 'real', [None], 'local'), Symbol)
+    assert isinstance(Symbol('a', 'real', [3], 'local'), Symbol)
+    assert isinstance(Symbol('a', 'real', [3, None], 'local'), Symbol)
+
+    # Test with invalid arguments
+    with pytest.raises(InternalError) as error:
+        Symbol('a', 'invalidtype', [], 'local')
+    assert ("Symbol can only be initialized with 'real' "
+            "or 'integer' datatypes.")in str(error.value)
+
+    with pytest.raises(InternalError) as error:
+        Symbol('a', 'real', [], 'invalidaccess')
+    assert ("Symbol access attribute can only be: 'local', 'external', "
+            "'read_arg', 'write_arg' or 'readwrite_arg'.") in str(error.value)
+
+    with pytest.raises(InternalError) as error:
+        Symbol('a', 'real', None, 'local')
+    assert "Symbol shape attribute must be a list." in str(error.value)
+
+    with pytest.raises(InternalError) as error:
+        Symbol('a', 'real', ['invalidshape'], 'local')
+    assert ("Symbol shape list elements can only be "
+            "'integer' or 'None'.") in str(error.value)
+
+
+def test_symbol_access_setter():
+    '''Test that a Symbol access can be set if given a new valid access
+    value, otherwise it raises a relevant exception'''
+
+    # Test with valid access value
+    sym = Symbol('a', 'real', [], 'local')
+    assert sym.access == 'local'
+    sym.access = 'read_arg'
+    assert sym.access == 'read_arg'
+
+    # Test with invalid access value
+    with pytest.raises(InternalError) as error:
+        sym.access = 'invalidaccess'
+    assert ("Symbol access attribute can only be: 'local', 'external', "
+            "'read_arg', 'write_arg' or 'readwrite_arg'.") in str(error.value)
+
+
+def test_symbol_can_be_printed():
+    '''Test that a Symbol instance can always be printed (i.e. is
+    initialised fully)'''
+    symbol = Symbol("sname", "real", [], "read_arg")
+    assert "sname<real, [], read_arg>" in str(symbol)
+
+
+# Test SymbolTable Class
+
+def test_symboltable_declare():
+    '''Test that the declare method inserts new symbols in the symbol
+    table, but raises appropiate errors when provied with wrong parameters
+    or duplate declarations'''
```

Comment:
s/duplate/duplicate/ and full stop please.

Reply:
Done